src/pages/Agregator/Agregator.component.tsx

```diff
@@ -2,66 +2,107 @@ import Box from '@mui/material/Box';
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
 
-import { utils } from 'ethers';
+import { BigNumber, utils } from 'ethers';
 import { useState, useEffect } from 'react';
 import { useForm } from 'react-hook-form';
 
-import { baseChains, ChainEnum, chains } from '../../config/chains';
+import { ChainType } from '../../config/chains';
 import { TokenTypeBase } from '../../config/tokens';
 import { PageView } from '../../components/PageView/PageView.component';
 import { ControlledCurrencyInput } from '../../components/CurrencyInput/CurrencyInput.controlled';
 import { ControlledInput } from '../../components/TextInput/TextInput.controlled';
 import { ControlledDropdown } from '../../components/Dropdown/Dropdown.controlled';
 import { ControlledInputWithButtonPillGroup } from '../../components/InputPillGroup/InputWithButtonPillGroup.controlled';
 
-import { AggregatorInputs } from './Agregator.fields';
+import { agregatorDefaultValues, AgregatorInputs } from './Agregator.fields';
 import {
   AgregatorComponentProps,
   AgregatorFormValues,
 } from './Agregator.types';
+import { mainnetPool } from '../../config/pools';
+import { AgregatorInfo } from './AgregatorInfo/AgregatorInfo.component';
 
 export const AgregatorComponent = ({
-  availableBalance,
   getTokenAvaliableBalance,
   getReceiveAmount,
   onSubmit,
 }: AgregatorComponentProps) => {
-  const [bassetOptions, setBassetOptions] = useState<TokenTypeBase[]>();
-  const [tokenDropdownDisabled, setTokenDropdownDisabled] = useState(true);
-
   const { handleSubmit, watch, setValue, control } =
     useForm<AgregatorFormValues>({
-      defaultValues: {
-        [AggregatorInputs.ChainDropdown]: '',
-        [AggregatorInputs.TokenDropdown]: '',
-        [AggregatorInputs.SendAmount]: '',
-        [AggregatorInputs.DestinationChain]: ChainEnum.RSK,
-        [AggregatorInputs.ReceiveAmount]: '',
-        [AggregatorInputs.ReceiveAddress]: '',
-      },
+      defaultValues: agregatorDefaultValues,
     });
-  const watchChain = watch(AggregatorInputs.ChainDropdown);
-  const watchToken = watch(AggregatorInputs.TokenDropdown);
-  const watchAmount = watch(AggregatorInputs.SendAmount);
+  const watchStartingChain = watch(AgregatorInputs.StartingChain);
```

Comment:
That's a lot of logic in this component. What do you think about moving part of it to the custom hooks?